### PR TITLE
Feature/variance reduction

### PR DESCRIPTION
**Pull request by marleenlki**
added docstrings and more details to readme

### CODE REVIEW

File: src/DecisionTreeAndRandomForest.jl [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module DecisionTreeAndRandomForest
 
+using DocStringExtensions
+using StatsBase: mode, sample, mean, countmap
```
Comment (cedric-len) [inline src/DecisionTreeAndRandomForest.jl:4]: We now have this line twice

Comment (cedric-len) [inline src/DecisionTreeAndRandomForest.jl:4]: This line now exists twice

File: src/RandomForest.jl [hunk 2]
```diff
@@ -69,9 +84,23 @@ function predict(forest::RandomForest, data::AbstractMatrix)
...
+- `forest::RandomForest`: The RandomForest to be printed.
+
+## Returns
+- `Nothing`: This function prints the structure of the `RandomForest`.
+"""
 function Base.show(io::IO, forest::RandomForest)
     for (index, tree) in enumerate(forest.trees)
         println(io, "Tree $index")
         print(tree)
+        println(io, "Tree $index")
```
Comment (cedric-len) [inline src/RandomForest.jl]: This also got duplicated

**Review by cedric-len — CHANGES_REQUESTED** [review]
Looks really good. Some lines just got duplicated in two places.

File: src/RandomForest.jl (hunk 2 above)
Comment (marleenlki) [reply src/RandomForest.jl]: Okay, thank you, I will look into it again :) Merging with the docstrings was a little painful

**Review by cedric-len — APPROVED** [review]

**Comment by cedric-len** [comment]
Fine for me :)

**Pull request merged**